### PR TITLE
MAINT: added cutpoint for overriding SecurityList implementation.

### PR DESCRIPTION
**Pull request by fawce**

### CODE REVIEW

File: zipline/utils/security_list.py [hunk 1]
```diff
@@ -102,6 +81,30 @@ def update_current(self, effective_date, symbols, change_func):
             change_func(sid)
 
 
+class SecurityListSet(object):
+
+    def __init__(self, current_date_func, lookup_func=None):
+        # provide a cut point to substitute other security
+        # list implementations.
+        self.sl_constructor = SecurityList
```
Comment (llllllllll) [inline zipline/utils/security_list.py]: If the idea is to make this configurable, could we make this an argument that defaults to SecurityList?

Comment (fawce) [inline zipline/utils/security_list.py]: idea is to give subclasses an override - seems confusing to put it in the arg list.

Comment (llllllllll) [inline zipline/utils/security_list.py]: if that is the case, should it be a class level attribute?

File: zipline/utils/security_list.py [hunk 2]
```diff
@@ -102,6 +83,30 @@ def update_current(self, effective_date, symbols, change_func):
             change_func(sid)
 
 
+class SecurityListSet(object):
+
+    def __init__(self, current_date_func, lookup_func=None):
+        # provide a cut point to substitute other security
+        # list implementations.
+        self.security_list_type = SecurityList
```
Comment (ssanderson) [inline zipline/utils/security_list.py]: Agreed with @llllllllll that this would be cleaner as a class-level attribute.

**Pull request merged**